src/varity/ref_gene.clj

```diff
@@ -213,22 +213,49 @@
   ([chr pos rgidx]
    (seek-gene-region chr pos rgidx nil))
   ([chr pos rgidx name]
-   ;; TODO seek intron region
-   ;; TODO seek UTR-5 or UTR-3 region
    (->> (if name
           (ref-genes name rgidx)
           (ref-genes chr pos rgidx))
         (map (fn [rg]
-               (let [exon-ranges ((case (:strand rg)
-                                    :forward identity
-                                    :reverse reverse) (:exon-ranges rg))
-                     idx (->> exon-ranges
-                              (keep-indexed (fn [i [s e]] (if (<= s pos e) i)))
-                              first)]
-                 {:exon-index (if idx
-                                (inc idx) ; 1-origin
-                                nil)
-                  :exon-count (count exon-ranges)
+               (let [sgn (case (:strand rg)
+                           :forward identity
+                           :reverse reverse)
+                     exon-ranges (sgn (:exon-ranges rg))
+                     exon-idx (->> exon-ranges
+                                   (keep-indexed (fn [i [s e]] (if (<= s pos e) i)))
+                                   first)
+                     intron-ranges (if exon-idx nil
+                                       (->> (:exon-ranges rg)
+                                            ((fn [r]
+                                               (loop [intron-ranges []
+                                                      left (first r)
+                                                      right (second r)
+                                                      ranges (rest r)]
+                                                 (if (empty? (first ranges)) intron-ranges
+                                                     (recur (conj intron-ranges [(inc (second left)) (dec (first right))])
+                                                            right
+                                                            (second ranges)
+                                                            (rest ranges))))))
```

Comment:
Could you define as an independent function such as `exon-ranges->intron-ragnges`? Additionally, please add relevant testing codes for it.